### PR TITLE
feat: add scripts option to lookup table

### PR DESCRIPTION
**Pull request by richardlau**
If set, "scripts" specifies the scripts from the module's package.json
to run instead of the default "test".

Allows `nan` to be added (second commit).

##### Checklist

- [x] `npm test` passes
- [x] tests are included
- [x] documentation is changed or added
- [x] contribution guidelines followed
      [here](https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by richardlau** [comment]
CI for this PR running `citgm nan`: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/68/

File: lib/package-manager/test.js [hunk 1]
```diff
@@ -66,44 +66,71 @@ function test(packageManager, context, next) {
...
+      : (context.module.scripts ? context.module.scripts : ['test']).map(
+          (x) => [packageManagerBin, 'run', x]
+        );
+    context.emit(
+      'data',
+      'silly',
+      `${context.module.name} npm-test:`,
+      `Scripts to execute: ${JSON.stringify(scripts)}.`
+    );
+    context.scriptsIter = scripts.values();
```
Comment (richardlau) [inline lib/package-manager/test.js]: `Array.values()` doesn't exist in Node.js 6 and 8. Will fix.

**Comment by richardlau** [comment]
> 
> 
> CI for this PR running `citgm nan`: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/68/

Rerun with fixes for Node.js 6 and 8 (avoid use of `array.values()`):
https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/69/console

**Comment by targos** [comment]
@SimenB Is that also what you need for Jest?

**Comment by SimenB** [comment]
Yes, perfect!

File: lib/package-manager/test.js [hunk 2]
```diff
@@ -66,44 +66,72 @@ function test(packageManager, context, next) {
...
-    proc.stdout.on('data', (data) => {
-      context.testOutput.append(data);
-      if (context.module.stripAnsi) {
-        data = stripAnsi(data.toString());
-        data = data.replace(/\r/g, '\n');
+    const scripts = context.options.customTest
+      ? [[context.options.customTest]]
+      : (context.module.scripts ? context.module.scripts : ['test']).map(
+          (x) => [packageManagerBin, 'run', x]
+        );
```
Comment (BridgeAR) [inline lib/package-manager/test.js]: Nit: wouldn't it be better to read if it's written as:

```js
let scripts;
if (context.options.customTest) {
  scripts = [[context.options.customTest]];
} else if (context.module.scripts) {
  scripts = context.module.scripts
    .map((script) => [packageManagerBin, 'run', script]);
} else {
  scripts = [packageManagerBin, 'run', 'test'];
}
```

Comment (richardlau) [reply lib/package-manager/test.js]: Fixed. FWIW `prettier` formats it slightly differently:
```diff
diff --git a/lib/package-manager/test.js b/lib/package-manager/test.js
index ba89636..39ae5b6 100644
--- a/lib/package-manager/test.js
+++ b/lib/package-manager/test.js
@@ -66,11 +66,18 @@ function test(packageManager, context, next) {

     /* Run `npm/yarn test`, or `/path/to/customTest.js` if the customTest option
      was passed */
-    const scripts = context.options.customTest
-      ? [[context.options.customTest]]
-      : (context.module.scripts ? context.module.scripts : ['test']).map(
-          (x) => [packageManagerBin, 'run', x]
-        );
+    let scripts;
+    if (context.options.customTest) {
+      scripts = [[context.options.customTest]];
+    } else if (context.module.scripts) {
+      scripts = context.module.scripts.map((script) => [
+        packageManagerBin,
+        'run',
+        script
+      ]);
+    } else {
+      scripts = [[packageManagerBin, 'test']];
+    }
     context.emit(
       'data',
       'silly',
```

**Comment by targos** [comment]
@richardlau can we go ahead with this?

**Comment by richardlau** [comment]
I released a patch release just before landing this. Currently waiting for Travis CI to complete on the rebased https://github.com/nodejs/citgm/pull/706 and then I'll land that too and release a new minor.

**Pull request merged**